src/main/software/amazon/event/ruler/MachineComplexityEvaluator.java

```diff
@@ -90,19 +93,26 @@ int evaluate(ByteState state) {
             }
         }
 
-        return maxSize;
+        return Math.max(nextNameStateMaxSize, maxSize);
     }
 
     /**
-     * Creates and returns a map of SingleByteTransition to all the wildcard patterns accessible from the
-     * SingleByteTransition. The map includes all SingleByteTransitions accessible from ByteState state. This function
-     * is O(N), where N is the number of states accessible from ByteState state.
+     * Populates a map of SingleByteTransition to all the wildcard patterns accessible from the SingleByteTransition.
+     * The map includes all SingleByteTransitions accessible from ByteState state. Also recursively evaluates the
+     * complexity of all other machines accessible through any next NameStates from this machine's matches. This
+     * function is O(N), where N is the number of states accessible from ByteState state, plus the time to recursively
+     * evaluate all other machines, which caps out below O(N^2) for each machine.
      *
-     * @param state Create a map containing all SingleByteTransitions accessible from this state.
-     * @return A map of SingleByteTransition to all the wildcard patterns accessible from the SingleByteTransition.
+     * Note: this function has two purposes. Although we could split this into two separate functions, doing it this way
```

Comment:
curious how much does this save us?

Reply:
I just split it out. Judging by the unit test times, this wasn't saving us much (5-10% max, maybe nothing). Not a big surprise. I'll send a new revision with this split out.